src/com/google/javascript/jscomp/ReplaceMessages.java

```diff
@@ -211,40 +209,44 @@ private void updateFunctionNode(JsMessage message, Node functionNode)
   private static Node constructAddOrStringNode(Iterator<CharSequence> partsIterator,
                                                Node argListNode)
       throws MalformedException {
-    CharSequence part = partsIterator.next();
-    Node partNode = null;
-    if (part instanceof JsMessage.PlaceholderReference) {
-      JsMessage.PlaceholderReference phRef =
-          (JsMessage.PlaceholderReference) part;
+    if (partsIterator.hasNext()) {
```

Comment:
Seems like we could avoid extra nesting (and make the diff smaller) by making the bailout condition return early.